nncf/experimental/post_training/algorithms/quantization/min_max/algorithm.py

```diff
@@ -0,0 +1,373 @@
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from copy import deepcopy
+from typing import Dict
+from typing import List
+from typing import Set
+from typing import TypeVar
+from typing import Union
+
+from nncf.common.graph.definitions import NNCFGraphNodeType
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.graph.transformations.commands import TargetPoint
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.common.hardware.config import HWConfigType
+from nncf.common.insertion_point_graph import InsertionPointGraph
+from nncf.common.quantization.quantizer_propagation.solver import QuantizerPropagationSolver
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizationPoint
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizerSetup
+from nncf.common.quantization.structs import QuantizableWeightedLayerNode
+from nncf.common.quantization.structs import QuantizationMode
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.common.tensor_statistics.collectors import TensorStatisticCollectorBase
+from nncf.common.utils.backend import BackendType, get_backend
+from nncf.common.utils.logger import logger as nncf_logger
+from nncf.common.utils.ordered_enum import OrderedEnum
+
+from nncf.experimental.post_training.algorithms import Algorithm
+from nncf.experimental.post_training.algorithms import AlgorithmParameters
+from nncf.experimental.post_training.algorithms.algorithm import PostTrainingAlgorithms
+from nncf.experimental.post_training.algorithms.quantization.min_max.backend import ALGO_BACKENDS
+from nncf.experimental.post_training.algorithms.quantization.min_max.utils import \
+    calculate_activation_quantizer_parameters
+from nncf.experimental.post_training.algorithms.quantization.min_max.utils import \
+    calculate_weight_quantizer_parameters
+from nncf.experimental.post_training.api.engine import Engine
+from nncf.experimental.post_training.factories import NNCFGraphFactory
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPoint
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPointsContainer
+
+ModelType = TypeVar('ModelType')
+
+
+class Preset(OrderedEnum):
+    PERFOMANCE = 'perfomance'
+    MIXED = 'mixed'
+    ACCURACY = 'accuracy'
+
+
+class Granularity(OrderedEnum):
+    PERTENSOR = 'pertensor'
+    PERCHANNEL = 'perchannel'
```

Comment:
If `Preset` and `Granularity` are not minmax-specific, consider separating them into an own file such as `nncf/experimental/post_training/algorithms/quantization/definitions.py`. Will help avoid circular imports in the future.

Reply:
These options are MinMaxQuantization specific since only this algorithm places Quantize-Dequantize pairs during Post-Training quantization.

Reply:
Aren't they actually common for all quantization, not just the `MinMax` type of quantization?

Reply:
Replaced these into the separate `definitions.py`.